subsys/net/lib/sockets/sockets_select.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/time.h>
+
+#include <net/socket.h>
+
+/* Get size, in elements, of an array within a struct. */
+#define STRUCT_MEMBER_ARRAY_SIZE(type, field) ARRAY_SIZE(((type *)0)->field)
+
+/* Outputs results in predefined word_idx and bit_mask vars. */
+#define FD_SET_CALC_OFFSETS(set) { \
+	unsigned int b_idx = fd % (sizeof(set->bitset[0]) * 8); \
+	word_idx = fd / (sizeof(set->bitset[0]) * 8); \
```

Comment:
Could the word_idx be passed as a parameter to the macro, it is easier to review the code as the reader can see that the word_idx is really used by the function like ZSOCK_FD_ISSET().

Reply:
> Could the word_idx be passed as a parameter to the macro

It's not just `word_idx`, it's also `bit_mask`, the boring questions like in which position these params go. That's why I did it like that initially. Ok, changing...